Release 4.1.0

aboedo opened the pull request:
- Added a new sample project available at `/Examples/WeatherApp`
    https://github.com/RevenueCat/react-native-purchases/pull/212
- iOS: 
    - Added a new method `setSimulatesAskToBuyInSandbox`, that allows developers to test deferred purchases easily.
- Bumped purchases-hybrid-common to 1.6.1 [Changelog here](https://github.com/RevenueCat/purchases-hybrid-common/releases/1.6.1)
- Bumped purchases-ios to 3.10.6 [Changelog here](https://github.com/RevenueCat/purchases-ios/releases/3.10.6)
- Bumped purchases-android to 4.0.4 [Changelog here](https://github.com/RevenueCat/purchases-hybrid-common/releases/4.0.4)
    https://github.com/RevenueCat/react-native-purchases/pull/216
- Fixed issues with CI builds in Android
    https://github.com/RevenueCat/react-native-purchases/pull/214

### Requirements: 
- [X] ~`purchases-hybrid-common` release 1.6.1~

vegaro commented on CHANGELOG.md:
```diff
@@ -1,7 +1,15 @@
-## 4.0.1
```
This shouldn't be removed right?

aboedo replied:
oops, not sure how that happened. good catch.

aboedo replied:
fixed

vegaro commented:
I have no idea what's up with CI

aboedo commented:
I'm trying to figure it out. 
So far what happened was:
- I updated our dependencies in general, so they no longer work with Xcode 11 and old cocoapods versions
- then I updated the Xcode version, which fixed some issues
- then there were some cocoapods issues, where it tries to build for the wrong platform because Xcode 12 changed the way you specify architectures
- after fixing that, we get this weird error where the precompiled header can't be found for some reason. 
I'm currently digging into that one. it's been fun.

aboedo commented on example/ios/Podfile:
```diff
@@ -71,9 +71,9 @@ target 'ReactNativeSample' do
   # Note that if you have use_frameworks! enabled, Flipper will not work and
   # you should disable these next few lines.
-  add_flipper_pods!
+  # add_flipper_pods!
```
I ended up commenting flipper out. 
It looks like it's not even recommended anymore in the most recent version. 

I tried upgrading, but auto-upgrade totally breaks. I think we should eventually kill this example altogether and start a new one, so that we can start with the latest RN version. we can just port our files over. 
https://react-native-community.github.io/upgrade-helper/?from=0.62.2&to=0.64.0

aboedo commented:
finally green
